pkg/instance/instance.go

```diff
@@ -230,21 +230,16 @@ func (i *Instance) SetInstanceType(instanceType InstanceType) {
 }
 
 // SetImage sets the image of the instance.
-// When calling in state 'Started', make sure to call AddVolume() before.
-// It is only allowed in the 'None' and 'Started' states.
+// It is not allowed in the 'Committed' and 'Started' states.
 func (i *Instance) SetImage(ctx context.Context, image string) error {
-	if !i.IsInState(StateNone, StateStarted) {
+	if i.IsInState(StateCommitted, StateStarted) {
 		return ErrSettingImageNotAllowed.WithParams(i.state.String())
 	}
 
 	if i.isSidecar {
 		return ErrSettingImageNotAllowedForSidecarsStarted
 	}
 
-	if i.state == StateStarted {
-		return i.setImageWithGracePeriod(ctx, image, nil)
-	}
-
```

Comment:
Removing this means we can't use `SetImage` anymore to replace an already running Instance with a new image.

Reply:
As using the function `SetImage` in two different cases might be confusing, we might want to consider creating another function like `UpgradeImage`, which allows the use case of changing the image.